Nadim.CinemaReservationSystem.Web/Services/CinemaService.cs

```diff
@@ -0,0 +1,311 @@
+            dbContext.Entry(changedCinema).Property("DefaultSeatPrice").IsModified = true;
+
+            dbContext.SaveChanges();
+
+            return new Result
+            {
+                ResultOk = true,
+            };
+        }
+
+        private Result ChangeCinemaVipSeatPrice(string name, decimal price) {
+            var changedCinema = dbContext.Cinemas.FirstOrDefault(c => c.Name == name);
+
+            if (changedCinema.VipSeatPrice == price && price < 0)
+            {
+                return new DataValidationResult
+                {
+                    ResultOk = false,
+                    Details = "Name is the same",
+                };
+            }
+
+            changedCinema.VipSeatPrice = price;
+            dbContext.Entry(changedCinema).Property("VipSeatPrice").IsModified = true;
+
+            dbContext.SaveChanges();
+
+            return new Result
+            {
+                ResultOk = true,
+            };
+        }
+
+        private Result ChangeCinemaRooms(string name, List<CinemaRoomCreationInfo> cinemaRooms)
+        {
+            var changedCinema = dbContext.Cinemas.FirstOrDefault(c => c.Name == name);
+
+            if (changedCinema == null)
+            {
+                return new DataValidationResult
+                {
+                    ResultOk = false,
+                    Details = "Cinema is not available."
+                };
+            }
+
+            var deleteCinemaRooms = dbContext.CinemaRooms.Include(room => room.Seats).Where(room => room.CinemaId == changedCinema.CinemaId);
+
+            if (deleteCinemaRooms.Count() != 0)
+            {
+
+                foreach (var room in deleteCinemaRooms)
+                {
+                    dbContext.Remove(room);
+                }
+                dbContext.SaveChanges();
+            }
+
+            List<CinemaRoom> newCinemaRooms = new List<CinemaRoom>();
+
+            foreach (CinemaRoomCreationInfo room in cinemaRooms)
+            {
+                newCinemaRooms.Add(new CinemaRoom
+                {
+                    Name = room.Name,
+                    Seats = new List<Seat>(),
+                });
+
+                foreach (SeatCreationInfo seat in room.Seats)
+                {
+                    newCinemaRooms.Last().Seats.Add(new Seat
+                    {
+                        Row = seat.Row,
+                        Column = seat.Column,
+                        Type = seat.Type,
+                    });
+                }
+            }
+
+            changedCinema.CinemaRooms = newCinemaRooms;
+
+            foreach (CinemaRoom room in changedCinema.CinemaRooms)
+            {
+                dbContext.CinemaRooms.Add(room);
+            }
+
+            dbContext.SaveChanges();
+            
+            return new Result
+            {
+                ResultOk = true,
+            };
+
+        }
+
+        public Result EditCinema(CinemaEditingInfo cinemaInfo)
```

Comment:
Can you think of a simpler and more efficient implementation of editing functional?